### PR TITLE
Ensure dags/ exists on generated-sql and speed up CI

### PR DESCRIPTION
**Pull request by scholtzan**
Okay, I'm trying to solve the mystery around `dags/` not being in `generated-sql` once and for all. I wrote a CI test step to check the files exist in: https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/12979/workflows/3dad7a02-d518-49b7-b5e1-39cd6482ab02/jobs/98704
The problem was that `generate-diff` overwrote the `generated-sql` contents.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -374,23 +374,20 @@ jobs:
...
               - << pipeline.git.tag >>
           steps:
+            - *attach_generated_sql
+            - *copy_generated_sql
             - docker/check:
                 docker-password: DOCKER_PASS
                 docker-username: DOCKER_USER
-      - docker/build: &public-image
-          image: ${CIRCLE_PROJECT_USERNAME+$CIRCLE_PROJECT_USERNAME/}${CIRCLE_PROJECT_REPONAME:-bigquery-etl}
-          tag: ${CIRCLE_TAG:-latest}
```
Comment (scholtzan) [inline .circleci/config.yml:390]: It seems unnecessary to build the docker image for every PR. It also takes around 5 minutes. It should be fine just doing it on main or am I missing something here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -461,22 +458,19 @@ jobs:
...
+            - *attach_generated_sql
+            - run:
+                name: Move generated-sql into place
+                command: |
+                  rm -rf sql/
+                  cp -r /tmp/workspace/private-generated-sql/sql sql
             - gcp-gcr/gcr-auth
-      - gcp-gcr/build-image: &private-image
-          image: bigquery-etl
-          tag: ${CIRCLE_TAG:-latest}
```
Comment (scholtzan) [inline .circleci/config.yml:476]: Same as `deploy`, just building when the deploy condition is true should be sufficient

File: .circleci/config.yml [hunk 3]
```diff
@@ -507,10 +501,15 @@ jobs:
             ./script/generate_sql \
               --target-project moz-fx-data-shared-prod
 
+            ./script/generate_airflow_dags
+
```
Comment (scholtzan) [inline .circleci/config.yml:505]: Adding diff for generated DAGs

File: .circleci/config.yml (hunk 1 above)
Comment (jklukas) [reply .circleci/config.yml:390]: I assume the intent here was to catch if we've introduced any errors that will affect the stability of building the image. I agree it's a pretty expensive step that would be nice to skip, though.

**Review by jklukas — APPROVED** [review]
r+wc

I don't have strong feelings about whether the docker build steps should stay in or be taken out. But that decision should probably be made in a separate PR. I'd tag @relud for input on that decision.

File: .circleci/config.yml (hunk 1 above)
Comment (relud) [reply .circleci/config.yml:390]: klukas has it right. I'm fine with this change.

**Review by relud — APPROVED** [review]

**Pull request merged**